use quote in backticks rather than escape (#1111)
neo4j/docs-cypher: modules/ROOT/pages/styleguide.adoc

RETURN "Cypher's a nice language", 'Mats\' quote: "statement"'
----

> * Avoid having to use backticks to quote characters and keywords.

Reviewer: ```suggestion
* Avoid using characters and keywords that require the input to be quoted.
```
Feels a little bit more actionable than "avoid having the db force a rule on you"
